desktop/core/src/desktop/middleware.py

```diff
@@ -944,15 +946,27 @@ def __call__(self, request):
     Rewrites the proxy headers so that only the most
     recent proxy is used.
     """
+    location = -1
+
+    if 'HTTP_X_REAL_IP' in request.META:
+      if 'HTTP_X_FORWARDED_FOR' in request.META and \
+        request.META['HTTP_X_REAL_IP'] in request.META['HTTP_X_FORWARDED_FOR']:
+        location = 0
+        for item in request.META['HTTP_X_FORWARDED_FOR'].split(','):
+          if request.META['HTTP_X_REAL_IP'].strip() != item.strip():
+            location += 1
+          else:
+            request.META['HTTP_X_FORWARDED_FOR'] = item.strip()
+            break;
+
```

Comment:
Please add a unit test or a descriptive example in the comments of what this code section is doing.
I think unit test would be the best!

Reply:
Thanks for the feedback. Added unit tests.